src/rdydata.c

```diff
@@ -43,6 +43,14 @@ static PetscErrorCode CheckBoundaryNumEdges(RDy rdy, const PetscInt boundary_ind
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+static PetscErrorCode CheckRegionIndex(RDy rdy, const PetscInt region_index) {
+  PetscFunctionBegin;
+  PetscCheck(region_index <= rdy->num_regions, rdy->comm, PETSC_ERR_USER,
```

Comment:
I think the predicate here should be `region_index < rdy->num_regions`, since the region's index is 0-based.

Reply:
Good catch.